Auth0/ASAuthenticationSession.swift

```diff
@@ -0,0 +1,52 @@
+// ASWebAuthenticationSession.swift
+//
+// Copyright (c) 2018 Auth0 (http://auth0.com)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import UIKit
+import AuthenticationServices
+
+#if swift(>=3.2)
+@available(iOS 12.0, *)
+class ASAuthenticationSession: AuthSession {
```

Comment:
Do we really need a whole new class or could we `typealias` the correct method based upon iOS?

Reply:
I think the class is probably justified as the ASWebAuthenticationSession or SFAuthenticationSession is stored in a member variable of the respective class. Then we have to worry about the error type in the handler callback which is related to the corresponding session type

Reply:
I have moved the changes inside the original SafariAuthenticationSession class